infra: address flake8 findings in webdriver/

thiagowfx opened the pull request:
Bug: https://github.com/web-platform-tests/rfcs/issues/171

thiagowfx commented:
Unit tests are (correctly!) failing. We'll need to fix the lint issues alongside this PR in order for them to pass.

thiagowfx commented:
Now there's only one error remaining.

```
webdriver/tests/bidi/input/perform_actions/invalid.py:495:1: F811 redefinition of unused 'test_params_pointer_action_up_down_button_missing' from line 416
```

@jrandolf please help here: which function should be renamed, and to what?

thiagowfx commented:
Added classic/ as well.

whimboo commented:
> test_params_pointer_action_up_down_button_missing

You have to change this one because it doesn't use a pointer up/down but a pointer move action:

https://github.com/web-platform-tests/wpt/blob/f9264d0700c76d662e829d3b6c55590d9e307a8f/webdriver/tests/bidi/input/perform_actions/invalid.py#L395-L397

thiagowfx commented:
Change to what? I was hoping to do the rename in a single round-trip review.
I am arbitrarily choosing `test_params_pointer_action_move_invalid`.

whimboo commented:
> Change to what? I was hoping to do the rename in a single round-trip review. I am arbitrarily choosing `test_params_pointer_action_move_invalid`.

It should be `test_params_pointer_action_move_coordinate_missing` because we remove a specific coordinate for the `move` pointer action.

jgraham commented on tools/flake8.ini:
```diff
@@ -10,14 +10,15 @@ select = E,W,F,N
 # E303: too many blank lines (3)
 # E305: expected 2 blank lines after end of function or class
 # E402: module level import not at top of file
+# E501: line too long
```
I don't think we want to ignore this for non-WebDriver files? I think you should be able to use [`per-file-ignores`](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-per-file-ignores) to just apply this to files under `webdriver` for now?

whimboo replied:
I would also like to leave that in for webdriver tests. Is there a reason why those long lines should be ignored?

thiagowfx replied:
The reason for ignoring is that there are too many violations of "line too long" in `webdriver/`. I did not want to make the diff even noisier than it already is.

I suppose `per-file-ignores` is a good compromise here. We could trivially fix all "line too long" occurrences later on  once we agree upon a formatter.

whimboo replied:
Would for an initial run maybe black help here? We could run it once before this PR gets merged. Usually we obeyed the 100 chars line length, so it's interesting to hear that many files are failing.

thiagowfx replied:
Done.

There were many occurrences, but they were mostly in a small number of files. I just formatted those files. It seems all E501 violations are now fixed.

jgraham commented:
I didn't check all the code changes in detail, please call out any where you aren't sure about the change.

thiagowfx commented:
> I didn't check all the code changes in detail, please call out any where you aren't sure about the change.

OK, I'll make comments in the changes that introduce a behavior change.

thiagowfx commented on webdriver/tests/bidi/browsing_context/navigation_started/navigation_started.py:
```diff
@@ -403,6 +402,8 @@ async def on_event(method, data):
         },
     )
 
+    remove_listener()
```
behavior change

thiagowfx commented on webdriver/tests/classic/perform_actions/support/refine.py:
```diff
@@ -8,7 +8,7 @@ def get_events(session):
             key = e["key"]
             hex_suffix = key[key.index("+") + 1:]
-            e["key"] = unichr(int(hex_suffix, 16))
+            e["key"] = chr(int(hex_suffix, 16))
```
behavior change (`unichr` was removed in python 3)

thiagowfx replied:
but this is supposedly a no-op

thiagowfx commented on webdriver/tests/classic/release_actions/support/refine.py:
```diff
@@ -8,7 +8,7 @@ def get_events(session):
             key = e["key"]
             hex_suffix = key[key.index("+") + 1:]
-            e["key"] = unichr(int(hex_suffix, 16))
+            e["key"] = chr(int(hex_suffix, 16))
```
ditto

thiagowfx commented:
@jgraham PTAL

whimboo commented on webdriver/tests/bidi/script/call_function/user_activation.py:
```diff
@@ -7,37 +7,35 @@
     result = await bidi_session.script.call_function(
-        function_declaration="() => document.body.appendChild(document.createTextNode('test')) && document.execCommand('selectAll') && document.execCommand('copy')",
+        function_declaration=
+        "() => document.body.appendChild(document.createTextNode('test')); document.execCommand('selectAll'); document.execCommand('copy')",
```
This change most likely caused the tests to fail. Why did you replace `&&` with `;`?

thiagowfx replied:
To make it fit into the max number of lines...(the cutoff was ~4 characters or so)

thiagowfx replied:
Fixed

whimboo replied:
But the `;` breaks the test. See the results at https://wpt.fyi/results/webdriver/tests/bidi/script/call_function/user_activation.py?diff&filter=ADC&run_id=6265633395703808&run_id=6331982050492416.

thiagowfx replied:
It's reverted now, I just split the string with '+' as an alternative

whimboo replied:
Can you please make sure the test doesn't fail locally? It's still broken due to: `FAIL message: webdriver.bidi.modules.script.ScriptEvaluateResultException: SyntaxError: missing ) in parenthetical`.

thiagowfx replied:
Where do you see this? I cannot see it in the logs of the stability jobs.

whimboo replied:
These are the wpt.fyi results for each browser. Just scroll down the list of jobs. Also you should always check those jobs even they appear green. Test failures aren't making those jobs fail.

thiagowfx replied:
> Also you should always check those jobs even they appear green. Test failures aren't making those jobs fail.

This is surprising. I usually only check the failing jobs. In this case, the stability jobs:

- https://github.com/web-platform-tests/wpt/pull/43096/checks?check_run_id=18682046616
- https://github.com/web-platform-tests/wpt/pull/43096/checks?check_run_id=18682046897

And none of the log there has failures pertaining to this PR (that I see).

Is this the only broken test, or are there more? I'd appreciate if you could tell me how to get the full list.

I understand how to get to the wpt.fyi job: https://github.com/web-platform-tests/wpt/pull/43096/checks?check_run_id=18684017796

From here, where do you go next?

thiagowfx replied:
It's fixed now!

The mistake was: I split the string with `+` but forgot to change `; -> &&`, my bad.

whimboo replied:
> Is this the only broken test, or are there more? I'd appreciate if you could tell me how to get the full list.
> 
> I understand how to get to the wpt.fyi job: https://github.com/web-platform-tests/wpt/pull/43096/checks?check_run_id=18684017796

As you can see on that page and switching between Chrome and Firefox the common failure was for `user_activation.py`. But it's also `perform_actions/wheel.py` which seems to maybe sporadically fail in Chrome. You can use the [Visual comparison of the results](https://wpt.fyi/results/webdriver/tests/bidi/script/call_function/user_activation.py?diff=&filter=ADC&run_id=5099264931266560&run_id=5187860208025600)] link right under the tables to actually get to the wpt.fyi results where you then can [extend the details for failures and see everything](https://wpt.fyi/results/webdriver/tests/bidi/input/perform_actions/wheel.py?diff&filter=ADC&run_id=5132912409706496&run_id=6289216054493184). And yes, there is a bug as well for `wheel.py`.

thiagowfx replied:
<img width="312" alt="image" src="https://github.com/web-platform-tests/wpt/assets/2840106/19bc4156-5370-4a6b-8530-12a6d37225a2">

thiagowfx replied:
And thanks for the explanation!

whimboo replied:
Oh I see but it's indeed broken: `FAIL message: webdriver.bidi.modules.script.ScriptEvaluateResultException: SyntaxError: Unexpected token '{'`. But yes, let me fix that separately given that this is already broken and cannot be found by the linter.

thiagowfx replied:
Feel free to push it on top of this PR. I am not planning to make any other changes to it.

whimboo replied:
I'll fix it via https://bugzilla.mozilla.org/show_bug.cgi?id=1864871.

whimboo reviewed (changes requested):
Whatever gets changed the tests should not start failing.

thiagowfx commented:
If there are no further comments, I believe this is ready to go.

thiagowfx commented:
@whimboo @jgraham could you please re-review? I'd ask an admin to merge this already, but the "changes requested" status blocks it

whimboo commented:
@web-platform-tests/admins can you please admin merge? The seen failures are not related to this particular change. Thanks.

gsnedders commented on tools/tox.ini:
```diff
@@ -15,7 +15,7 @@ changedir =
 commands =
   !flake8-!mypy: pytest --cov=tools --cov-report=term {posargs}
-  flake8: flake8 --append-config={toxinidir}/flake8.ini {posargs}
+  flake8: flake8 --append-config={toxinidir}/flake8.ini {toxinidir} {toxinidir}/../webdriver {posargs}
```
I think I'd like to see a bit more consensus on https://github.com/web-platform-tests/rfcs/issues/171 prior to landing this. We don't often add new lint rules for tests, _and_ this is a different approach.

Currently, all the test lints are run via `./wpt lint`, and only tooling is linted here. This is especially notable because we only run the tools unittests when there are changes in `tools`, so we wouldn't even run this on most WebDriver PRs.

I also wouldn't expect this to be run in downstream repos, because by and large they're only running `wpt lint` and not the full tooling CI.

thiagowfx replied:
@jgraham @foolip @DanielRyanSmith (as infra folks):

Any thoughts here?

thiagowfx replied:
I also have another ask: In case there's no resolution of this in, say, 2-3 business days, then let's revert the `tox.ini` change and land the rest.

I think we can all agree that the lint fixes here are uncontroversial and valuable.

The contentious part seems to merely be how to enforce the linting in the infra.

gsnedders replied:
Yeah, landing the rest is totally non-objectionable, I think. The question is how we want to enforce it.

Commit:
infra: address flake8 findings in webdriver/

Rename duplicate test to test_params_pointer_action_move_coordinate_missing

Bug: https://github.com/web-platform-tests/rfcs/issues/171
Cmdline: flake8 --append-config=tools/flake8.ini webdriver/tests/bidi                                                                                                                                                 :

Commit:
revert flake8 webdriver command in tools/tox.ini

thiagowfx commented:
@DanielRyanSmith and/or @jgraham could you admin merge? This is addressing the linter issues, without enforcing the linter. The enforcement discussion is still WIP at https://github.com/web-platform-tests/rfcs/issues/171
